projects/plugins/jetpack/modules/likes/queuehandler.js

```diff
@@ -236,40 +236,12 @@ function JetpackLikesMessageListener( event ) {
 				}
 
 				const element = document.createElement( 'li' );
-				// Appending element before fetching avatar_URL to avoid racing conditions on the list order
 				list.append( element );
 
-				try {
-					const response = await fetch( liker.avatar_URL, { method: 'HEAD' } );
-					if ( ! response.ok ) {
-						// Image doesn't exist, remove the element
-						element.remove();
-						return;
-					}
-				} catch ( error ) {
-					// Error occurred, but we check if there is a default image to be used in case the error were CORS
-					const url = new URL( liker.avatar_URL );
-					const defaultImageUrl = decodeURI( url.searchParams.get( 'd' ) );
-					if ( defaultImageUrl ) {
-						const defaultImageResponse = await fetch( defaultImageUrl, { method: 'HEAD' } );
-						if ( defaultImageResponse.ok ) {
-							liker.avatar_URL = defaultImageUrl;
-						} else {
-							// Default image doesn't exist, remove the element
-							element.remove();
-							return;
-						}
-					} else {
-						// Error occurred while checking image existence, remove the element
-						element.remove();
-						return;
-					}
-				}
-
```

Comment:
Just noting here that the original thinking behind introducing this section was to keep `projects/plugins/jetpack/modules/likes/queuehandler.js` logic in sync to what we already have on the WPCOM side (since it has its own `queuehandler.js` from the times before we started working on the Like block project).

Here's related patch that introduced some of the changes to WPCOM: D133753-code.

From my perspective, we can skip fetching if we are ok that WPCOM side will work slightly differently than the JP side. Alternatively, we could remove fetching from the WPCOM side as well.

Reply:
>From my perspective, we can skip fetching if we are ok that WPCOM side will work slightly differently than the JP side. Alternatively, we could remove fetching from the WPCOM side as well.

Why would we need fetching at .com? Jetpack features should work as closely the same across both, without divergence in code because it becomes a pain to maintain otherwise.

Reply:
> Why would we need fetching at .com?

The fetch was introduced in D109839-code in May 2023 to check if the image exists.

What I think we could do is to look deeper why I certain image would not exist.